TypeInjections/Tool/Tool.fs

```diff
@@ -21,91 +48,61 @@ module Tool =
-            Utils.log "***** remote github"
 
-            let path = remote.[3]
-            let pathOld = $"{path}/Old"
-            let pathNew = $"{path}/New"
-            let url = remote.[0]
-            let commit1 = remote.[1]
-            let commit2 = remote.[2]
+        let commit1, commit2 = git.oldHash, git.newHash
+        let path = git.outputFolder
+        let pathOld = $"{path}/Old"
+        let pathNew = $"{path}/New"
 
-            for a in [ pathOld; pathNew ] do
-                if Directory.Exists(a) then
+        // lib2git needs an empty folder to store the repo clones
+        for a in [ pathOld; pathNew ] do
+            if Directory.Exists(a) then
+                // if user lets us delete existing folder
+                if git.force then
                     Directory.Delete(a, true)
+                // else break program
+                else
+                    failwith $"There is already an {a}, we need empty folder to place git repo. Chose different output folder or chose --force to automatically overwrite data"
 
-            checkoutCommit commit1 url pathOld
-            checkoutCommit commit2 url pathNew
+        checkoutCommit commit1 location pathOld
+        checkoutCommit commit2 location pathNew
 
+        Utils.log $"{pathOld}, {pathNew}, {path}/Output"
 
-            pathOld, pathNew, $"{path}/Output"
 
-        else
-            failwith "unable to read flags"
-            
-    // organize arguments with the flag they associated with 
-    let run opt =
-        let local = opt.local |> Seq.toList
-        let git = opt.git |> Seq.toList
-        let remote = opt.remote |> Seq.toList
-        // given the list of arguments associated with each flag, run typeCart
-        let oldFolder, newFolder, outputFolder = interpretFlags local git remote
+    let runLocal (local: LocalOptions) : unit =
+
+        Utils.log "***** local project comparison"
+        Utils.log $"{local.oldFolder}, {local.newFolder}, {local.outputFolder}"
 
-        Utils.log $"old folder is {oldFolder}, new folder is {newFolder}, output folder is {outputFolder}"
 
     [<EntryPoint>]
     let main (argv: string array) =
 
         Utils.log "***** Entering Tool.fs"
 
         let result =
-            Parser.Default.ParseArguments<options>(argv)
+            Parser.Default.ParseArguments<LocalOptions, GitOptions> argv
 
-        //TODO get a better understanding of result
         match result with
-        | :? (Parsed<options>) as parsed -> run parsed.Value
-        | :? (NotParsed<'a>) as notParsed -> Utils.log $"Here is non parsed arguments {notParsed}"
-        | _ as unknown -> Utils.log $"don't understand this argument {unknown}"  
+
+        | :? (Parsed<obj>) as command ->
+            match command.Value with
+            | :? LocalOptions as opts -> runLocal opts
+            | :? GitOptions as opts -> runGit opts
+
+        | :? (NotParsed<obj>) as value-> Utils.log $"finally got to not parsed {value}"
```

Comment:
We should print out a usage page you would get from a CLI tool when you run it with `--help` or so. We should also provide such an option, i.e., `typecart --help` that prints the same page

Reply:
This is built in feature of `CommandLine.Parser` All the `HelpText` is what gets written out when user asks for help
Ex. `typecart git --help`
Ouptut:

```
  -o, --old           Required. commit hash of old version

  -n, --new           Required. commit hash of new version

  -p, --output        Required. folder path to place typeCart files

  -e, --entrypoint    specify filename you want to run typecart on

  -c, --clone         Give a git URL

  -f, --force         Overwrite the New and Old folders in Output folder

  --help              Display this help screen.

  --version           Display version information.
```

Reply:
I noticed the capitalization is off, will fix shortly

Reply:
Fixed